### PR TITLE
CMake changes to allow building without lusol

### PR DESCRIPTION
**Pull request by cameronrutherford**
My editor screwed with formatting. I can force push to this PR with minimal changes once I disable CMake formatting in my editor...

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,39 +13,47 @@ project(ReSolve VERSION "0.99.1")
...
   message(FATAL_ERROR "In-source build prohibited.")
 endif()
 
-option(RESOLVE_TEST_WITH_BSUB "Use `jsrun` instead of `mpirun` commands when running tests" OFF)
-option(RESOLVE_USE_KLU  "Use KLU, AMD and COLAMD libraries from SuiteSparse" ON)
-option(RESOLVE_USE_LUSOL "Build the LUSOL library" ON)
+option(RESOLVE_TEST_WITH_BSUB
+       "Use `jsrun` instead of `mpirun` commands when running tests" OFF)
+option(RESOLVE_USE_KLU "Use KLU, AMD and COLAMD libraries from SuiteSparse" ON)
+option(RESOLVE_USE_LUSOL "Build the LUSOL library" OFF)
```
Comment (cameronrutherford) [inline CMakeLists.txt:27]: Off by default...

File: tests/unit/matrix/CMakeLists.txt [hunk 2]
```diff
@@ -16,18 +16,30 @@ target_link_libraries(runMatrixHandlerTests.exe PRIVATE ReSolve resolve_matrix)
...
+target_link_libraries(runMatrixFactorizationTests.exe PRIVATE ReSolve
+                                                              resolve_matrix)
 
 # Build LUSOL-related tests
-add_executable(runLUSOLTests.exe runLUSOLTests.cpp)
-target_link_libraries(runLUSOLTests.exe PRIVATE ReSolve resolve_matrix)
+if(RESOLVE_USE_LUSOL)
+  add_executable(runLUSOLTests.exe runLUSOLTests.cpp)
+  target_link_libraries(runLUSOLTests.exe PRIVATE ReSolve resolve_matrix)
+endif()
```
Comment (cameronrutherford) [inline tests/unit/matrix/CMakeLists.txt:26]: Small changes needed in testing to disable LUSOL...

File: tests/unit/matrix/CMakeLists.txt [hunk 3]
```diff
@@ -16,18 +16,30 @@ target_link_libraries(runMatrixHandlerTests.exe PRIVATE ReSolve resolve_matrix)
...
-
-add_test(NAME matrix_test               COMMAND $<TARGET_FILE:runMatrixIoTests.exe>)
-add_test(NAME matrix_handler_test       COMMAND $<TARGET_FILE:runMatrixHandlerTests.exe>)
-add_test(NAME matrix_factorization_test COMMAND $<TARGET_FILE:runMatrixFactorizationTests.exe>)
-add_test(NAME lusol_test                COMMAND $<TARGET_FILE:runLUSOLTests.exe>)
+set(installable_tests runMatrixIoTests.exe runMatrixHandlerTests.exe
+                      runMatrixFactorizationTests.exe)
+if(RESOLVE_USE_LUSOL)
+  list(APPEND installable_tests runLUSOLTests.exe)
+endif()
```
Comment (cameronrutherford) [inline tests/unit/matrix/CMakeLists.txt:33]: And here

File: tests/unit/matrix/CMakeLists.txt [hunk 4]
```diff
@@ -16,18 +16,30 @@ target_link_libraries(runMatrixHandlerTests.exe PRIVATE ReSolve resolve_matrix)
...
+
+add_test(NAME matrix_test COMMAND $<TARGET_FILE:runMatrixIoTests.exe>)
+add_test(NAME matrix_handler_test
+         COMMAND $<TARGET_FILE:runMatrixHandlerTests.exe>)
+add_test(NAME matrix_factorization_test
+         COMMAND $<TARGET_FILE:runMatrixFactorizationTests.exe>)
+
+if(RESOLVE_USE_LUSOL)
+  add_test(NAME lusol_test COMMAND $<TARGET_FILE:runLUSOLTests.exe>)
+endif()
```
Comment (cameronrutherford) [inline tests/unit/matrix/CMakeLists.txt:45]: And here

**Comment by pelesh** [comment]
> My editor screwed with formatting. I can force push to this PR with minimal changes once I disable CMake formatting in my editor...

Please do. Otherwise this is good to merge. Thank you!

**Comment by pelesh** [comment]
I believe we can close this PR because this issue was addressed in #118. Feel free to reopen, if needed.

**Pull request closed without merging**